script/schema_doc.py

```
@@ -0,0 +1,910 @@
+
+def json_get(name):
+    if name == "core":
+        name = "esphome"
+
+    json_doc = json_docs.get(name)
+    if json_doc:
+        return json_doc
+
+    json_file_name = os.path.join(args.schema_dir, name + ".json")
+    if os.path.exists(json_file_name):
+        f = open(json_file_name, "r", encoding="utf-8-sig")
```

Ruff (0.12.2)

Use a context manager for opening files

(SIM115)